module-rt/tests/multiple.rs

```diff
@@ -0,0 +1,213 @@
+        }
+
+        let pizza_box_n = rng.gen_range(10, 100);
+        let mut pizza_boxes: Vec<Box<dyn PizzaBox>> = Vec::new();
+        for _ in 0..pizza_box_n {
+            pizza_boxes.push(self.pizza_stores.choose(&mut rng).unwrap().wrap_pizza_box().unwrap_import().into_proxy())
+        }
+        self.pizza_boxes = pizza_boxes;
+
+        Vec::new()
+    }
+}
+
+fn execute_module<M: UserModule + 'static>(args: Vec<String>) {
+    fmoudle_rt::start::<Intra, M>(args);
+}
+
+fn create_module(
+    mut ctx: ExecutorContext<Intra, PlainThread>,
+    n: usize,
+) -> (ExecutorContext<Intra, PlainThread>, RtoContext, Arc<RwLock<dyn fmoudle_rt::coordinator_interface::FoundryModule>>)
+{
+    // we use n-1 since we don't prepare a service for its own.
+    let exports: Vec<(String, Vec<u8>)> = (0..n - 1).map(|_| ("".to_owned(), vec![])).collect();
+    let (transport_send, transport_recv) = ctx.ipc.take().unwrap().split();
+    let config = RtoConfig::default_setup();
+    let (rto_context, module): (_, ServiceToImport<dyn FoundryModule>) =
+        remote_trait_object::Context::with_initial_service_import(config, transport_send, transport_recv);
+    let module: Arc<RwLock<dyn FoundryModule>> = module.into_proxy();
+
+    module.write().initialize(&[], &exports);
+    (ctx, rto_context, module)
+}
+
+#[test]
+fn multiple() {
+    let mut module_names = Vec::new();
+    let n = 10;
+    for _ in 0..n {
+        let name = generate_random_name();
+        add_function_pool(name.clone(), Arc::new(execute_module::<ModuleA>));
+        module_names.push(name);
+    }
+
+    let mut modules = Vec::new();
+    for name in module_names {
+        let executor = execute::<Intra, PlainThread>(&name).unwrap();
+        modules.push(create_module(executor, n));
+    }
+
+    // link and bootstrap
+    for i in 0..n {
+        for j in 0..n {
+            if i >= j {
+                continue
+            }
+
+            let port_name = generate_random_name();
+
+            let mut port1: Box<dyn Port> = modules[i].2.write().create_port(&port_name).unwrap_import().into_proxy();
+            let mut port2: Box<dyn Port> = modules[j].2.write().create_port(&port_name).unwrap_import().into_proxy();
+            let (ipc_arg1, ipc_arg2) = Intra::arguments_for_both_ends();
+
+            let join = std::thread::spawn(move || {
+                port1.initialize(PartialRtoConfig::from_rto_config(RtoConfig::default_setup()), ipc_arg1, true);
+                port1
+            });
+            port2.initialize(PartialRtoConfig::from_rto_config(RtoConfig::default_setup()), ipc_arg2, true);
+            let mut port1 = join.join().unwrap();
+
+            let handles_1_to_2 = port1.export(&[if j > i {
+                j - 1
+            } else {
+                j
+            }]);
+            let handles_2_to_1 = port2.export(&[if i > j {
+                i - 1
+            } else {
+                i
+            }]);
```

Comment:
Could you tell me why does this test need `i - 1`?

Reply:
Please check the comment